crates/bevy_reflect/src/type_info_stack.rs

```diff
@@ -0,0 +1,49 @@
+use crate::TypeInfo;
+use core::fmt::{Debug, Formatter};
+use core::slice::Iter;
+
+/// Helper struct for managing a stack of [`TypeInfo`] instances.
+///
+/// This is useful for tracking the type hierarchy when serializing and deserializing types.
+#[derive(Default, Clone)]
+pub(crate) struct TypeInfoStack {
+    stack: Vec<&'static TypeInfo>,
+}
+
+impl TypeInfoStack {
+    /// Create a new empty [`TypeInfoStack`].
+    pub const fn new() -> Self {
+        Self { stack: Vec::new() }
+    }
+
+    /// Push a new [`TypeInfo`] onto the stack.
+    pub fn push_info(&mut self, type_info: &'static TypeInfo) {
```

Comment:
Ah, it's because I originally had a `push<T: Typed>` convenience method. I removed it though since it wasn't actually being used.

I'll update this method's name for consistency!